### PR TITLE
Fix the sort algorithm for sorting the query execution result by column

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Fix the sort algorithm for sorting the query execution result by column

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-graph/src/graph-manager/action/execution/ExecutionResult.ts [hunk 1]
```diff
@@ -144,13 +144,17 @@ export const getTDSRowRankByColumnInAsc = (
...
-  return Number(guaranteeNonNullable(a1)) - Number(guaranteeNonNullable(b1));
+  const a1 = a.values[colIndex];
+  const b1 = b.values[colIndex];
+  if (a1 === null || a1 === undefined) {
+    return -1;
+  }
+  if (b1 === null || b1 === undefined) {
+    return 1;
+  }
+  if (typeof a1 === 'string' && typeof b1 === 'string') {
```
Comment (MauricioUyaguari) [inline packages/legend-graph/src/graph-manager/action/execution/ExecutionResult.ts:155]: cant we just do .toString() then just localCompare on all of them.

Comment (YannanGao-gs) [reply packages/legend-graph/src/graph-manager/action/execution/ExecutionResult.ts:155]: tested this -- not working for number

Comment (YannanGao-gs) [reply packages/legend-graph/src/graph-manager/action/execution/ExecutionResult.ts:155]: "10" < "2"

Comment (MauricioUyaguari) [reply packages/legend-graph/src/graph-manager/action/execution/ExecutionResult.ts:155]: please see https://stackoverflow.com/questions/10198257/comparing-2-strings-alphabetically-for-sorting-purposes
`"a5b".localeCompare("a21b", undefined, { numeric: true })`
we do this when comparing versions in studio. 
We also have `isString` functions in shared.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**